Enable IOUPreview for bill splits

Julesssss opened the pull request:
CC @luacmartins @mountiny 

### Details

Enabled the IOUPreview to display within groups. Bill Splits amount, participant, comment, and currency data will show in the preview. Tapping it will open the group participants page for now. 

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/Expensify/issues/270707

### Tests

**1) Request money from a single user**
- Locate any user chat
- Request money from them via the FAB or ChatReport menu
- An IOUPreview should display
- It should show 'Cash' at the top of the component
- The amount and currency should be displayed
- If a description was added, it should display
- Both user avatars should be displayed
- Hovering over the avatars should display the participant emails
- Tapping it should take you to the IOUDetails Modal

**2) Split a bill with a group of users**
- Locate any group chat with 3 or more users
- Split a bill with the group via the FAB or ChatReport menu
- Multiple 1:1 requests should be created
- An IOUPreview should display within the group chat
- It should show 'Split' at the top of the component
- The amount and currency should be displayed
- If a description was added, it should display
- One avatar should be displayed, with a counter displaying the number of other participants
- Hovering over the avatars should display the participant emails
- Tapping it should take you to the Details page, showing the list of other participants

### Offline tests

Run the above tests. **[FOR TEST 1 ONLY]** In addition, you should see a 'Total will update when you're back online' message in the IOUPReview if your users are on different currencies. This doesn't apply to group splits as we don't have an associated iouReport to read from.
<img width="472" alt="Screenshot 2023-05-03 at 15 44 07" src="https://user-images.githubusercontent.com/10736861/235951335-1176ba59-a115-461c-a2bc-fb7122b0c4d5.png">


### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x]  I linked the correct issue in the `### Fixed Issues` section above
- [x]  I wrote clear testing steps that cover the changes made in this PR
    - [x]  I added steps for local testing in the `Tests` section
    - [x]  I added steps for the expected offline behavior in the `Offline steps` section
    - [x]  I added steps for Staging and/or Production testing in the `QA steps` section
    - [x]  I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x]  I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x]  I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x]  I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x]  I ran the tests on **all platforms** & verified they passed on:
    - [x]  Android / native
    - [x]  Android / Chrome
    - [x]  iOS / native
    - [x]  iOS / Safari
    - [x]  MacOS / Chrome / Safari
    - [x]  MacOS / Desktop
- [x]  I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x]  I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x]  I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x]  I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x]  I verified that comments were added to code that is not self explanatory
    - [x]  I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x]  I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x]  If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x]  I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x]  I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x]  I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x]  I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x]  If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x]  I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x]  I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x]  I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x]  I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x]  I verified that if a function's arguments changed that all usages have also been updated correctly
- [x]  If a new component is created I verified that:
    - [x]  A similar component doesn't exist in the codebase
    - [x]  All props are defined accurately and each prop has a `/** comment above it */`
    - [x]  The file is named correctly
    - [x]  The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x]  The only data being stored in the state is data necessary for rendering and nothing else
    - [x]  For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x]  Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x]  All JSX used for rendering exists in the render method
    - [x]  The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x]  If any new file was added I verified that:
    - [x]  The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x]  If a new CSS style is added I verified that:
    - [x]  A similar style doesn't already exist
    - [x]  The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x]  If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x]  If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x]  If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x]  If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x]  If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x]  I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<img width="747" alt="Screenshot 2023-05-03 at 15 35 11" src="https://user-images.githubusercontent.com/10736861/235948503-d326ac4d-7a4c-433f-b303-86b8bfa60eaf.png">


</details>

<details>
<summary>Mobile Web - Chrome</summary>

![Screenshot_20230503_165724](https://user-images.githubusercontent.com/10736861/235971470-2feb89c5-6475-426c-ad02-b720d4b3da1f.png)


</details>

<details>
<summary>Mobile Web - Safari</summary>

![Simulator Screen Shot - iPhone 14 - 2023-05-03 at 16 07 47](https://user-images.githubusercontent.com/10736861/235969396-10ddfb7c-6b54-495a-b6b2-d6f8ff5d5c78.png)

</details>

<details>
<summary>Desktop</summary>

![Screenshot 2023-05-03 at 15 36 09](https://user-images.githubusercontent.com/10736861/235948865-427e6a4c-1034-49ba-9d93-ea2f3b89545c.png)


</details>

<details>
<summary>iOS</summary>

![IMG_0133](https://user-images.githubusercontent.com/10736861/236450294-e2324e32-d582-4b61-a00f-15a5a556ddfe.PNG)


</details>

<details>
<summary>Android</summary>

![Screenshot_20230505-124738](https://user-images.githubusercontent.com/10736861/236450357-dfe07f64-50b5-41ea-a070-92f3f66eac9b.png)


</details>

Julesssss committed:
locate iouReportID in ReportActionItem to simplify onyx report retrieval

Julesssss committed:
pas requestReportID to IOUAction comp as string

Julesssss committed:
Merge branch 'main' into jules-iouPreview-showForBillSplit

Julesssss committed:
always show iouPreview component

Julesssss committed:
implement launching of the chat participants page from the IOUPreview

Julesssss committed:
IOUPreview now launches either the IOUDetailsPage or ParticipantsPage

Julesssss committed:
Merge branch 'main' into jules-iouPreview-showForBillSplit

Julesssss committed:
improve iouPreview callback naming

Julesssss committed:
for group split IOUPreview, use currency and total from action

Julesssss committed:
don't show IOUPreview paid icon for splits

Julesssss committed:
add isBillSplit prop to IOUPreview to selectively show content

Julesssss committed:
show correct amount for split IOUPreview

Julesssss committed:
remove unused formatted 'send $amount' string

Julesssss committed:
rename split amount string format

Julesssss committed:
display IOU type in IOUPreview component

Julesssss committed:
refactor and simplify iou format strings

Julesssss committed:
replace temporary isGroupSplit condition

Julesssss committed:
refactor getAvatars to OptionListUtils helper function

Julesssss committed:
document OptionListUtils helper function and rename

Julesssss committed:
display split participant avatars in IOUPreview

Julesssss committed:
simplify IOUPreview participant avatar logic

Julesssss committed:
remove 'x paid y' from IOUPreview

Julesssss committed:
display IOU description in IOUPreview

Julesssss committed:
re-enable pending conversation message

Julesssss committed:
show the IOUPreview component for all IOUActions

Julesssss committed:
clean up iouReportID logic

Julesssss committed:
always show the clickable IOUPreview styles

Julesssss committed:
show tooltip for split IOUPreview component

Julesssss committed:
Merge branch 'main' into jules-iouPreview-showForBillSplit

Julesssss committed:
remove unnecessary props and const from IOUPreview

Julesssss committed:
apply formatting fixes

Julesssss commented on src/components/MoneyRequestConfirmationList.js:
```diff
@@ -104,7 +104,7 @@ class MoneyRequestConfirmationList extends Component {
     getSplitOrRequestOptions() {
         return [{
-            text: this.props.translate(this.props.hasMultipleParticipants ? 'iou.split' : 'iou.request', {
+            text: this.props.translate(this.props.hasMultipleParticipants ? 'iou.splitAmount' : 'iou.requestAmount', {
```
A minor change that was necessary. Else the new 'split' string would need an alternative name.

Julesssss committed:
update missed iouRequest translation

Julesssss commented:
Sorry @danieldoglas & @pecanoro, Pullerbear was assigned twice. Please feel free to ignore this review

Julesssss commented:
Bumping for review, this is a blocker for many Manual Request/EC3 project issues. 

@pecanoro @parasharrajat @mountiny @luacmartins

pecanoro commented on src/components/ReportActionItem/IOUPreview.js:
```diff
@@ -136,32 +130,32 @@ const IOUPreview = (props) => {
     const sessionEmail = lodashGet(props.session, 'email', null);
     const managerEmail = props.iouReport.managerEmail || '';
     const ownerEmail = props.iouReport.ownerEmail || '';
+    const participantEmails = props.isBillSplit ? props.action.originalMessage.participants : [managerEmail, ownerEmail];
```
Just a question since I am not sure how the IOU action is built. When it's a 1:1 action, we don't store the managerEmail and ownerEmail in participants? Just wondering why we can't use participants for both cases.

mountiny replied:
I agree with the conclusion from Jules

pecanoro replied:
Thank you for the explanation!

pecanoro commented on src/components/ReportActionItem/IOUPreview.js:
```diff
@@ -136,32 +130,32 @@ const IOUPreview = (props) => {
-        ) : '';
-    const avatarTooltip = [Str.removeSMSDomain(managerEmail), Str.removeSMSDomain(ownerEmail)];
+    let cachedTotal;
+    if (props.isBillSplit) {
+        if (props.action && props.action.originalMessage && props.action.originalMessage.currency) {
+            cachedTotal = props.numberFormat(
+                Math.abs(props.action.originalMessage.amount) / 100,
+                {style: 'currency', currency: props.action.originalMessage.currency},
+            );
+        } else {
+            cachedTotal = props.numberFormat(Math.abs(props.action.originalMessage.amount) / 100);
+        }
+    } else {
+        cachedTotal = props.iouReport.total && props.iouReport.currency
+            ? props.numberFormat(
+                Math.abs(props.iouReport.total) / 100,
+                {style: 'currency', currency: props.iouReport.currency},
+            ) : '';
+    }
```
Can you leave a comment explaining why we get the amount from different sources depending on the type of IOU? For example, I don't exactly what's the difference between `props.iouReport.total` and `props.action.originalMessage.amount`.

pecanoro replied:
Ohh I see later that you left a comment saying "// Only 1:1 requests require an iouReport. For splits we rely on reportAction data.", I think you leave a comment saying the same here.

pecanoro replied:
Either way, instead of this complex if/else in which the code is almost repeated, I think you can simply find the proper total depending on the request type and then maybe call `props.numberFormat` once with the amount.

luacmartins replied:
Agree with only calling numberFormat once

Julesssss replied:
Thanks both, I vastly improved and simplified this logic.

pecanoro commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -178,9 +178,16 @@ class ReportActionItem extends Component {
         let children;
         if (this.props.action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU) {
+            // Only 1:1 requests require an iouReport. For splits we rely on reportAction data.
+            let iouReportID = '0';
+            if (this.props.action.originalMessage.IOUReportID) {
+                iouReportID = this.props.action.originalMessage.IOUReportID.toString();
+            }
```
You can use `?:` for this, right? Not sure if I am not seeing something here.

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg  | https://18143.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://18143.pr-testing.expensify.com)  |

luacmartins commented on src/components/ReportActionItem/IOUPreview.js:
```diff
@@ -136,32 +130,32 @@ const IOUPreview = (props) => {
-            {style: 'currency', currency: props.iouReport.currency},
-        ) : '';
-    const avatarTooltip = [Str.removeSMSDomain(managerEmail), Str.removeSMSDomain(ownerEmail)];
+    let cachedTotal;
```
Can we change this name? cachedTotal is really misleading since we also have a cachedTotal key in the iouReport object (which we should really get rid of)

Julesssss replied:
Done

luacmartins commented on src/languages/en.js:
```diff
@@ -316,13 +317,8 @@ export default {
-        paidyou: ({manager}) => `${manager} paid you`,
-        split: ({amount}) => `Split ${amount}`,
-        send: ({amount}) => `Send ${amount}`,
+        requestAmount: ({amount}) => `Request ${amount}`,
+        splitAmount: ({amount}) => `Split ${amount}`,
```
We are changing these to start with lower case letters

luacmartins commented on src/languages/es.js:
```diff
@@ -315,13 +316,8 @@ export default {
-        paidyou: ({manager}) => `${manager} te pagó`,
-        split: ({amount}) => `Dividir ${amount}`,
-        send: ({amount}) => `Enviar ${amount}`,
+        requestAmount: ({amount}) => `Solicitar ${amount}`,
+        splitAmount: ({amount}) => `Dividir ${amount}`,
```
Same here. Start with lower case please

luacmartins reviewed (changes requested):
Looking good. Left a few comments

Julesssss committed:
simplify reportActionItem logic for detecting iouReportId and improve comment

Julesssss committed:
uncapitalize the IOUPreview messages

Julesssss committed:
improve and simplify amount formatting logic

Julesssss committed:
Merge branch 'main' into jules-iouPreview-showForBillSplit

luacmartins reviewed (dismissed):
LGTM

Julesssss committed:
apply lint fixes and improve IOUPreview documentation

Julesssss commented:
Thanks, both. I applied changes to all feedback excluding the [iou action logic](https://github.com/Expensify/App/pull/18143/files#r1184792193). Please take a look and share any further concerns!

Julesssss commented:
I'll add the iOS/Android screenshots tomorrow morning

Julesssss committed:
remove trailing space

pecanoro commented:
Will review it tomorrow morning again so we can merge asap!

mountiny reviewed (commented):
Code looks good to me but havent tests, @aimane-chnaif and @parasharrajat should be bale to handle that

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg  | https://18143.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://18143.pr-testing.expensify.com)  |

mountiny commented:
@aimane-chnaif I believe all these changes do not require any backend changes.

Julesssss commented on src/components/MoneyRequestConfirmationList.js:
```diff
@@ -104,7 +104,7 @@ class MoneyRequestConfirmationList extends Component {
     getSplitOrRequestOptions() {
         return [{
-            text: this.props.translate(this.props.hasMultipleParticipants ? 'iou.split' : 'iou.request', {
+            text: this.props.translate(this.props.hasMultipleParticipants ? 'iou.splitAmount' : 'iou.requestAmount', {
```
Ah thanks. We're reusing that in two places so I'll need to duplicate the string.

Julesssss replied:
Done

luacmartins commented:
@aimane-chnaif we'll have some capitalization issues for now. Also, the `with <users>` part is no longer part of the template so those are expected for now. I think the only real issue is https://github.com/Expensify/App/pull/18143#discussion_r1185346558

luacmartins commented:
@Julesssss we have some conflicts

Julesssss committed:
merge main and resolve conflicts with new currency util function

Julesssss committed:
capitalise request/split string in certain cases to avoid duplicate string

Julesssss commented:
Thanks @aimane-chnaif, nice finds. 

I fixed the capitalization issue, fixed conflicts, and fixed another bug where phone number users weren't working properly.

As for the split participant avatars not showing, I tried to resolve this but it's a bit tricky to figure out the correct format for the optimistic data. I created a [separate follow-up issue](https://github.com/Expensify/App/issues/18472) to work on this as it probably shouldn't block the changes.

Julesssss commented:
ready for your review: @aimane-chnaif @pecanoro @luacmartins 

I think [this thread](https://github.com/Expensify/App/pull/18143#discussion_r1184792193) is the last remaining blocker.

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18143/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18143/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg  | https://18143.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18143/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://18143.pr-testing.expensify.com)  |

mountiny commented:
@Julesssss i am still getting the issue wiht the total amount showing in both of the previews 
<img width="978" alt="image" src="https://user-images.githubusercontent.com/36083550/236496461-be0d510a-8e32-46cd-b67f-842f52d3f937.png">

mountiny reviewed (dismissed):
great progress

Julesssss commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -93,30 +96,28 @@ const IOUAction = (props) => {
+                ]}
                 isHovered={props.isHovered}
             />
-            {shouldShowIOUPreview && (
```
Thanks, yeah I think this was also related to a currency conflict from yesterday. I had already fixed this in my next PR but will fix this now 👍

Julesssss committed:
revert back to showing a single IOUPreview

Julesssss committed:
simplify logic and resolve 'line too long' lint error

Julesssss commented:
Okay, sorry about that. Completely removing the `shouldDisplayIOUPreview` logic should have been saved for the next PR. 

Fixed!

Julesssss commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -78,6 +78,8 @@ const IOUAction = (props) => {
         }
     };
 
+    const shouldShowIOUPreview = props.isMostRecentIOUReportAction || props.action.originalMessage.type === 'pay';
```
While we could include 'pay' in the array, it doesn't really make sense because it is not a IOURequest. Also, we're going to completely remove this condition with my next PR.

mountiny replied:
We can do that

mountiny reviewed (changes requested):
~requesting $90 in a group of 3 people has split the amount as if it was a group of 4 people~ all good I have added more people to the group than I wanted.... so its 4 actually
<img width="934" alt="image" src="https://user-images.githubusercontent.com/36083550/236510330-a6d874a7-9230-45be-9772-7ade5fa7b5bf.png">

mountiny reviewed (approved):
All good, I am idiot.

luacmartins commented:
Just waiting on final review by @aimane-chnaif! Thanks for the urgency on this one!

Julesssss commented:
Thanks for the detailed review @aimane-chnaif. You should claim the bug bounty for those other issues you found too 😉

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4921605794) to staging by https://github.com/luacmartins in version: 1.3.12-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4922047640) to production by https://github.com/roryabraham in version: 1.3.12-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4921605794) to staging by https://github.com/luacmartins in version: 1.3.12-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

eVoloshchak commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -93,21 +98,21 @@ const IOUAction = (props) => {
-                    iouReportID={props.action.originalMessage.IOUReportID.toString()}
+                    iouReportID={props.requestReportID}
                     chatReportID={props.chatReportID}
+                    isBillSplit={hasMultipleParticipants}
```
Coming from https://github.com/Expensify/App/issues/18671.
We were using participants.length to determine if action is a split action, so the check would fail if you delete all of the participants from the workspace.
This was fixed by checking the original type of the action
`const isSplitBillAction = lodashGet(props.action, 'originalMessage.type', '') === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;`
